### PR TITLE
Addresses #8970, Fixes DataGrid not processing all new items on change.

### LINKED ISSUE #8970: [DataGrid] AvaloniaList<T> implementation of InsertRange does not correctly invoke _collectionChanged
When trying to use `AvaloniaList<T>` and its `AddRange` method, I found that my `DataGrid` control only updated one element rather than the expected many. After looking into the source I believe I found the problem.

**Problem:**
`InsertRange` in `AvaloniaList<T>` implementation calls `NotifyAdd` which invokes the `_collectionChanged` event with `NotifyCollectionChangedAction.Add` which is meant for single item changes. `InsertRange` should instead be calling with `NotifyCollectionChangedAction.Reset` instead which is meant for large changes.

**Consequences:**
Anyone who attempts to use `AvaloniaList<T>` with the `InsertRange` method or more likely used `AddRange` method with a control that listens to `CollectionChanged` will not be updated correctly.

**Workaround:**
After calling `AddRange` or `InsertRange` use reflection to manually invoke the event with correct `NotifyCollectionChangedAction`.
```cs
var eventDelegate = typeof(AvaloniaList<T>).GetField("_collectionChanged", BindingFlags.Instance | BindingFlags.NonPublic)?.GetValue(this) as NotifyCollectionChangedEventHandler;
eventDelegate?.Invoke(this, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
```

**To Reproduce**

1. Create a `DataGrid` control in a window. (Using `DataGrid` as that is how I found out the issue and what I have experience with)
2. Bind an `AvaloniaList<T>` to Items.
3. Try update the list with `AddRange` or `InsertRange` using a collection with 2 or more elements.
4. Notice that only 1 element will be added.

**Expected behavior**
After using `AddRange` or `InsertRange` all items should be represented in the `DataGrid`

**Screenshots**
N/A

**Desktop:**
- OS: Linux (openSUSE Tumbleweed)
- Version: 0.10.18

### PR DESCRIPTION
**Pull request by ghasttear1**
## What does the pull request do?
The pull request makes it so that DataGridCollectionView properly handles multiple new items being added when handling a collection changed event. Before this change only the first item was taken `addedItem = args.NewItems?[0];` and processed. With this pull request I have changed the behaviour to iterate through `args.NewItems` and process each new item.
This issue has been brought up from my own personal issue #8970.

## What is the current behavior?
The current behaviour is that attempting to modify a collection that is bound to the DataGrid by adding multiple items will only result in the first item been added to the visible collection.

## What is the updated/expected behavior with this PR?
When adding multiple items to a collection that is bound to DataGrid, the view shows all items that have been added instead of only 1.

I have tested and confirmed that this pull request fixes the issue.

For testing I modified DataGridPage.xaml.cs of samples/ControlCatalog project with the following:
```diff
line 57:
-var collectionView3 = new DataGridCollectionView(items);
+var collectionView3 = new Avalonia.Collections.AvaloniaList<Person>(items);
line 62:
-addButton.Click += (a, b) => collectionView3.AddNew();
+addButton.Click += (a, b) => collectionView3.AddRange(items);
```
Run this on current master and confirm the incorrect behaviour.
Run this with the changes made in my pull request and confirm that this incorrect behaviour is now fixed.

## How was the solution implemented (if it's not obvious)?
Obvious, check diff

## Checklist
N/A

## Breaking changes
N/A

## Obsoletions / Deprecations
N/A

## Fixed issues
Fixes #8970

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=9135) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=9135) before we can accept your contribution.<br/><br/>:x: ghasttear1 [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=9135)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=9135) it.</sub>

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024456-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Controls.DataGrid/Collections/DataGridCollectionView.cs [hunk 1]
```diff
@@ -3424,11 +3423,17 @@ private void ProcessCollectionChanged(NotifyCollectionChangedEventArgs args)
             }
 
             // fire notifications for adds
-            if ((args.Action == NotifyCollectionChangedAction.Add ||
-                args.Action == NotifyCollectionChangedAction.Replace) &&
-                (Filter == null || PassesFilter(addedItem)))
+            if (args.NewItems != null && 
+                (args.Action == NotifyCollectionChangedAction.Add ||
+                 args.Action == NotifyCollectionChangedAction.Replace))
```
Comment (maxkatz6) [inline src/Avalonia.Controls.DataGrid/Collections/DataGridCollectionView.cs:3428]: Seems replace still won't work well when multiple items are replaced at the same time. 
Should be similar code be added above for ProcessRemoveEvent?

Comment (ghasttear1) [reply src/Avalonia.Controls.DataGrid/Collections/DataGridCollectionView.cs:3428]: I'll have a look later, I really was only looking to fix adding multiple items but you are likely right about that. I didn't know whether or not replacing had issues as I hadn't tried.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024471-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Controls.DataGrid/Collections/DataGridCollectionView.cs (hunk 1 above)
Comment (ghasttear1) [reply src/Avalonia.Controls.DataGrid/Collections/DataGridCollectionView.cs:3430]: I just pushed a new commit which adds similar changes to removing items and that has also fixed removing multiple items, I wasn't too sure how to test replacing multiple items. In the process I did unintentionally cause a merge commit for my PR by clicking a sync branch button on github out of curiosity, and I know that rebasing is preferred and I'm not sure if that is any issue or not...

Comment (maxkatz6) [reply src/Avalonia.Controls.DataGrid/Collections/DataGridCollectionView.cs:3430]: No, it's not an issue.
Thanks! I will check later today or tomorrow.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024472-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024707-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024788-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**